It is not possible to set avahi_runtime_dir using configure options.

--- a/configure.ac	2024-03-11 14:28:49.719377887 +0100
+++ b/configure.ac	2024-03-11 14:30:40.530644269 +0100
@@ -1004,7 +1004,7 @@
 #
 # Avahi runtime dir
 #
-avahi_runtime_dir="/run"
+avahi_runtime_dir="/var/run"
 avahi_socket="${avahi_runtime_dir}/avahi-daemon/socket"
 AC_SUBST(avahi_runtime_dir)
 AC_SUBST(avahi_socket)
